UsbAccess/UsbAccess.cpp

```diff
@@ -508,6 +563,172 @@ namespace Plugin {
         returnResponse(result);
     }
 
+    uint32_t UsbAccess::getPartitionCountWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        string command =  "lsblk -o NAME,MOUNTPOINT | grep \"sd[a-z]\" | grep \"/\" |wc -l";
+        FILE* fp = popen(command.c_str(), "r");
+        if (fp == NULL) 
+        {
+            result = false;
+        }   
+        
+        while (fgets(buffer, 32, fp) != NULL);
+
+        string PartitionCount = buffer;
+
+        int ret = pclose(fp);
+        if (ret == -1) 
+        {
+            result = false;
+        }
+        response["PartitionCount"] = PartitionCount;
+        returnResponse(result);
+    }
+
+
+
+    uint32_t UsbAccess::getPartitionInfoWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        unsigned long long int blocksize = 0;
+        JsonArray arr;
+        JsonObject ent;
+        string pathParam;
+        if (parameters.HasLabel("path"))
+            pathParam = parameters["path"].String();
+
+        struct statvfs fs;
+        if (statvfs(pathParam.c_str(), &fs) == -1) {
+            result = false;
+        }
+
+        string command_str1 = "df -T  ";
+        string command_str2 = " | tail -n +2 | awk '{print $2}'";
+        string command = command_str1 + pathParam + command_str2;
+        FILE* fp = popen(command.c_str(), "r");
```

Comment:
Since the popen is used on the data coming from outside, v_secure_popen() should be used for handling more secure way.

Reply:
Hi @srikanth-vv , 
I could not find a sample usage of **v_secure_popen()**. Could you please share a sample implementation?

Tried to use v_secure_popen() from code in lib32-libsyswrapper/6.1+gitAUTOINC+09b41f3324-r0/git/source/secure_wrapper.h

But observing "**Permission denied**" logs where this method was used it was added:
Jun 21 03:40:08 HP44HRDK WPEFramework[10113]: /run/media/sdb1: Permission denied

Also observed that the value of FileSystem in getPartitionInfoWrapper() method is returning empty. Can you please provide some inputs?